compiler/ghc/default.nix

```diff
@@ -65,6 +65,13 @@ let self =
   # necessary fix for iOS: https://www.reddit.com/r/haskell/comments/4ttdz1/building_an_osxi386_to_iosarm64_cross_compiler/d5qvd67/
   disableLargeAddressSpace ? stdenv.targetPlatform.isDarwin && stdenv.targetPlatform.isAarch64
 
+, useLdGold ? 
+    # might be better check to see if cc is clang/llvm?
+    # use gold as the linker on linux to improve link times
+    # do not use it on musl due to a ld.gold bug. See: <https://sourceware.org/bugzilla/show_bug.cgi?id=22266>.
+    (stdenv.targetPlatform.isLinux && !stdenv.targetPlatform.isAndroid && !stdenv.targetPlatform.isMusl) 
+    || stdenv.targetPlatform.isAarch32
```

Comment:
After looking closer at this, I believe this is wrong. We do not unconditionally enable this in aarch32.
I believe the `|| stdenv.targetPlatform.isAarch32` should go.